src/VisualStudio/Core/Def/Implementation/Remote/VisualStudioRemoteHostClientProvider.cs

```diff
@@ -46,8 +47,11 @@ internal sealed class Factory : IWorkspaceServiceFactory
             [Obsolete(MefConstruction.FactoryMethodMessage, error: true)]
             public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
             {
+                // We don't want to bring up the OOP process in a VS cloud environment client instance
+                // Avoids proffering brokered services on the client instance.
+                var isCloudEnvironmentClient = workspaceServices.GetRequiredService<IWorkspaceContextService>().IsCloudEnvironmentClient();
                 if (!RemoteHostOptions.IsUsingServiceHubOutOfProcess(workspaceServices)
-                    || workspaceServices.Workspace is not VisualStudioWorkspace)
+                    || workspaceServices.Workspace is not VisualStudioWorkspace || isCloudEnvironmentClient)
```

Comment:
Nit: This would be more readable (and not query the WorkspaceContextSerfvice if not needed)
```
if (!RemoteHostOptions.IsUsingServiceHubOutOfProcess(workspaceServices) ||
     workspaceServices.Workspace is not VisualStudioWorkspace || 
     workspaceServices.GetRequiredService<IWorkspaceContextService>().IsCloudEnvironmentClient())
```